scripts/apps/monitoring/controllers/AggregateCtrl.js

```diff
@@ -87,25 +90,56 @@ export function AggregateCtrl($scope, api, desks, workspaces, preferencesService
             });
         } else {
             return workspaces.getActiveId()
-                .then(function(activeWorkspace) {
-                    if (self.settings != null && self.settings.desk) {
-                        return deskMonitoringConfig(self.settings.desk);
-                    } else if (activeWorkspace.type === 'workspace') {
-                        return preferencesService.get(PREFERENCES_KEY)
-                            .then(function(preference) {
-                                if (preference && preference[activeWorkspace.id] && preference[activeWorkspace.id].groups) {
-                                    return {'type': 'workspace', 'groups': preference[activeWorkspace.id].groups};
+            .then(function(activeWorkspace) {
```

Comment:
This will become unmaintainable in it's current state, please refactor this to contain less `if` statements and please split it into smaller chunks by creating functions and documenting the flow of execution.